server/api/attendance/attendance.controller.js

```diff
@@ -0,0 +1,674 @@
+                    if (!smallgroup){return res.send(400, 'Incorrect day code');}
+                    // Small group, and not a bonus day
+                    else if (smallgroup.dayCode === code){
+                        // Check if the user already submitted a small group, non-bonus attendance
+                        if (submittedSmall){
+                            // if it is already submitted, return
+                            return res.send(400, 'Small group attendance already recorded: ' + submittedSmall.verified);
+                        }
+                        // if not, create the attendance object
+                        return Attendance.create({
+                            classYear: classYear._id,
+                            user: user._id,
+
+                            date: isoDateToTime(new Date()),
+                            datetime: new Date(),
+
+                            bonusDay: false,
+                            smallgroup: true,
+
+                            verified: !needsVerification,
+                            code: code,
+
+                        }, function(err,submission){
+                            return res.send(200, {'type':'Small group attendance', 'unverified': needsVerification});
+                        });
+
+                    }
+                    // Small group & bonus day
+                    else if (smallgroup.bonuDayCode === code){
+                        // Check if the user already submitted a small group & bonus attendance
+                        if (submittedSmallBonus){
+                            // if it is already submitted, return
+                            return res.send(400, 'Small group bonus attendance already recorded: ' + submittedSmallBonus.verified);
+                        }
+                        // if not, create the attendance object
+                        return Attendance.create({
+                            classYear: classYear._id,
+                            user: user._id,
+
+                            date: isoDateToTime(new Date()),
+                            datetime: new Date(),
+
+                            bonusDay: true,
+                            smallgroup: true,
+
+                            verified: !needsVerification,
+                            code: code,
+
+                        }, function(err,submission){
+                            return res.send(200, {'type':'Small group bonus attendance', 'unverified': needsVerification});
+                        });
+                    }
+                });
+            }
+        });
+    });
+};
+// *******************************************************
+
+
+// *******************************************************
+// Set attendance as present (no verification)
+// router.post('/attend/small/:user', auth.hasRole('mentor'), controller.setAttendanceSmall);
+// router.post('/attend/full:user/', auth.hasRole('mentor'), controller.setAttendanceFull);
```

Comment:
Its just a typo.... It is meant as documentation for the future. Thats why you are supposed to verify it.